cpp/src/arrow/chunked_array.h

```diff
@@ -116,6 +118,10 @@ class ARROW_EXPORT ChunkedArray {
   /// \return an ArrayVector of chunks
   const ArrayVector& chunks() const { return chunks_; }
 
+  /// \return The set of device allocation types used by the chunks in this
+  /// chunked array.
+  DeviceAllocationTypeSet DeviceTypeSet() const;
```

Comment:
Nit (maybe?) I would've expected this to be named `device_types()`

Reply:
Done.